src/NzbDrone.Core/Indexers/IndexerLimitService.cs

```diff
@@ -63,5 +64,33 @@ public bool AtQueryLimit(IndexerDefinition indexer)
 
             return false;
         }
+
+        public int CalculateRetryAfterDownloadLimit(IndexerDefinition indexer)
+        {
+            if (indexer.Id > 0 && ((IIndexerSettings)indexer.Settings).BaseSettings.GrabLimit.HasValue)
+            {
+                var firstHistorySince = _historyService.FindFirstSince(indexer.Id, DateTime.Now.AddHours(-24), new List<HistoryEventType> { HistoryEventType.ReleaseGrabbed });
+                if (firstHistorySince != null)
+                {
+                    return Convert.ToInt32(firstHistorySince.Date.AddHours(24).Subtract(DateTime.Now).TotalSeconds);
+                }
+            }
+
+            return 0;
+        }
+
+        public int CalculateRetryAfterQueryLimit(IndexerDefinition indexer)
+        {
+            if (indexer.Id > 0 && ((IIndexerSettings)indexer.Settings).BaseSettings.QueryLimit.HasValue)
+            {
+                var firstHistorySince = _historyService.FindFirstSince(indexer.Id, DateTime.Now.AddHours(-24), new List<HistoryEventType> { HistoryEventType.IndexerQuery, HistoryEventType.IndexerRss });
+                if (firstHistorySince != null)
+                {
+                    return Convert.ToInt32(firstHistorySince.Date.AddHours(24).Subtract(DateTime.Now).TotalSeconds);
```

Comment:
Instead of a hard coded 24 hours since last grab could this default to 24hrs and have an indexer override that's passable?

e.g. BTN is 150/hr, once the next hour hits all is fine.


and likely out of scope but future consideration - ensure indexer queries that hit the rate limit from the site pass that back with the appropriate retry after to the requesting app

Reply:
I don't think we want a setting for this if that's what you are saying

Reply:
Definition based, not user setting